### PR TITLE
Auto close the edit modal on form submit

### LINKED ISSUE #16347: Modal does not close
### Required System information

- Node.js version: v18
- NPM version: 9.5.0
- Strapi version: 4.9.0
- Database: SQLite
- Operating system: MacOS Ventura
- Is your project Javascript or Typescript: Typescript

### Describe the bug

When you configure the view of a entity you are usually allowed to rename the columns but when you click on `Finish` the modal stays visible. This create the illusion that it wasn't saved properly but if you click on the `x` you can see it works fine. It's just the modal that is not closing

### Steps to reproduce the behavior

1. Go to any entity, eg `users`
2. Click on the cog to configure the view
3. Try to edit a column's label
4. Click on Finish
5. Modal stays visible

### Expected behavior

Modal should close when you click on finish

### Screenshots
Screenshot 2023-04-11 at 1 39 40 PM

### Code snippets

N/A

### Additional context

If this is truly a bug, I'll be happy to create a PR fixing it.

### PR DESCRIPTION
**Pull request by LuisRodriguezLD**
### What does it do?

~It resets the reducer state so the edit modal can disappear once the form is submitted.~
Closes the modal after submitting
Added a new test to make sure it closes every time the form is submitted
~Also adjusted an existing test.~

### Why is it needed?

The edit modal is not auto closing and it gives the impression of not working

### How to test it?

1. Open any entity table, eg users
2. Click on the cog AKA configure view
3. Edit any label
4. Click on Finish
5. Modal should now close

### Related issue(s)/PR(s)

Closes #16347

### CODE REVIEW

File: packages/core/admin/admin/src/content-manager/pages/ListSettingsView/reducer.js [hunk 1]
```diff
@@ -68,6 +68,8 @@ const reducer = (state = initialState, action) =>
         break;
       }
       case 'SUBMIT_FIELD_FORM': {
+        draftState.fieldForm = {};
+        draftState.fieldToEdit = '';
```
Comment (markkaylor) [inline packages/core/admin/admin/src/content-manager/pages/ListSettingsView/reducer.js:72]: I noticed that there is already an action type 'UNSET_FIELD_TO_EDIT' that does exactly this.

I think the same result could be achieved by dispatching that action after this one. There is also already a handler for closing the modal that dispatches this action, so you could just call it.

https://github.com/strapi/strapi/blob/0fbd3d3b0c48fac09c1d93d351154e482f0c99e9/packages/core/admin/admin/src/content-manager/pages/ListSettingsView/index.js#L119-L130

becomes:
```js
 const handleCloseModal = () => { 
   dispatch({ 
     type: 'UNSET_FIELD_TO_EDIT', 
   }); 
 }; 
  
 const handleSubmitFieldEdit = (e) => { 
   e.preventDefault(); 
   dispatch({ 
     type: 'SUBMIT_FIELD_FORM', 
   }); 
	 handleCloseModal()
 }; 
```

That way this state reset remains defined in a single place. Wdyt?

**Review by markkaylor — COMMENTED** [review]
Good catch and thanks for the PR! I just have one comment and I'm curious if you considered the approach.

File: packages/core/admin/admin/src/content-manager/pages/ListSettingsView/reducer.js (hunk 1 above)
Comment (LuisRodriguezLD) [reply packages/core/admin/admin/src/content-manager/pages/ListSettingsView/reducer.js]: Personally, I think the change should be handled from the reducer but I'm all in for consistency.
Code has been updated. 
Thanks for taking the time to review this PR.

**Review by markkaylor — APPROVED** [review]
Thanks for the change, looks good to me 👍

File: packages/core/admin/admin/src/content-manager/pages/ListSettingsView/reducer.js (hunk 1 above)
Comment (markkaylor) [reply packages/core/admin/admin/src/content-manager/pages/ListSettingsView/reducer.js]: I agree that consistency is another good reason to do it this way. I've approved and will merge as is. I would still be interested in your reasoning for why this should be handled in the reducer though, if you care to share :)

**Pull request merged**